zerver/management/commands/delete_old_unclaimed_attachments.py

```diff
@@ -42,10 +51,33 @@ def handle(self, *args: Any, **options: Any) -> None:
                 f"* {old_archived_attachment.file_name} created at {old_archived_attachment.create_time}"
             )
 
-        print("")
+        if options["for_real"]:
+            do_delete_old_unclaimed_attachments(delta_weeks)
+            print("")
+            print("Unclaimed files deleted.")
+
+        if options["clean_up_storage"]:
+            print("")
+            self.clean_storage_backends(dry_run=not options["for_real"])
+
         if not options["for_real"]:
+            print("")
             raise CommandError("This was a dry run. Pass -f to actually delete.")
 
-        do_delete_old_unclaimed_attachments(delta_weeks)
-        print("")
-        print("Unclaimed files deleted.")
+    def clean_storage_backends(self, dry_run: bool = True) -> None:
+        print("Removing extra files in storage blackend")
+        to_delete = []
+        for path_id, modified_at in all_message_attachments():
+            if Attachment.objects.filter(path_id=path_id).exists():
+                continue
+            if ArchivedAttachment.objects.filter(path_id=path_id).exists():
+                continue
+            print(f"* {path_id} modified at {modified_at}")
+            if dry_run:
+                continue
+            to_delete.append(path_id)
+            if len(to_delete) > 1000:
+                delete_message_attachments(to_delete)
+                to_delete = []
+        if not dry_run and len(to_delete) > 0:
+            delete_message_attachments(to_delete)
```

Comment:
This may be unsafe to run often, in that the sequence in which we add files to the attachment storage I think adds them before the `Attachment` row gets created, and thus there's a race where we might incorrectly delete a file that was just uploaded. 

I'm not sure I see an easy way to fix that race other than re-checking the `to_delete` list after a 5s delay or something before doing the deletion.

Reply:
We have the modification timestamp on files as part of `all_message_attachments`, so we can just leave a 5 minute grace period for recent ones; repushed with that change.